### ISSUE TITLE
why does the only method that appears is socket?

### ISSUE THREAD

**Issue by Renoldi** [flutter/devtools#8380]
<-- Please describe your problem here. Be sure to include repro steps. -->
___
**DevTools version**: 2.37.3
**IDE**: VSCode
**Connected Device**:
CPU / OS: x64 (64 bit) android
Connected app type: Flutter native (debug build)
Dart Version: 3.5.3
Flutter Version: 3.24.3 / stable
Framework / Engine: 2663184aa7 / 36335019a8

**Comment by LiamMarega** [comment]
I have the same problem

**Comment by elliette** [comment]
@bkonyi Was this something you fixed with https://github.com/flutter/devtools/issues/3033? Your changes were included in the 2.37.2 release, so I would have expected this to be fixed.

**Comment by elliette** [comment]
@Renoldi Could you share a small repro with us and share whichever package you are using to send these requests? Thanks!

**Comment by bkonyi** [comment]
> @bkonyi Was this something you fixed with #3033? Your changes were included in the 2.37.2 release, so I would have expected this to be fixed.

This is probably WAI, but it depends on what package / library is being used to make HTTP requests. Making an HTTP request involves opening a socket, so these `SOCKET` entries are always expected when HTTP profiling is enabled. However, if the HTTP library being used isn't using `dart:io`'s `HttpClient` underneath, those requests won't appear in the profiler.

**Comment by saeedprogrammer** [comment]
same issue with dio
![Screenshot 2024-10-31 at 4 54 11 AM](https://github.com/user-attachments/assets/5b822417-1d3a-42be-8245-6cd93c4b5b63)

Flutter 3.24.4 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 603104015d (6 days ago) • 2024-10-24 08:01:25 -0700
Engine • revision db49896cf2
Tools • Dart 3.5.4 • DevTools 2.37.3

**Comment by zigapovhe** [comment]
@bkonyi I've have been using 
- http package v1.2.2
- Flutter v3.24.5
- Dart version 3.5.4
- DevTools version 2.37.3
- MacOS 15.1
- Tested on Android emulator, SDK 35

All I see is SOCKET requests.

**Comment by nicks258** [comment]
Any updates on this still facing the issue in `Flutter 3.24.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision dec2ee5c1f (3 weeks ago) • 2024-11-13 11:13:06 -0800
Engine • revision a18df97ca5
Tools • Dart 3.5.4 • DevTools 2.37.3
`

**Comment by Nico04** [comment]
Same here on several projects, all of which using classic `http` package (not `dio`)

**Comment by zigapovhe** [comment]
> > @bkonyi Was this something you fixed with #3033? Your changes were included in the 2.37.2 release, so I would have expected this to be fixed.
> 
> This is probably WAI, but it depends on what package / library is being used to make HTTP requests. Making an HTTP request involves opening a socket, so these `SOCKET` entries are always expected when HTTP profiling is enabled. However, if the HTTP library being used isn't using `dart:io`'s `HttpClient` underneath, those requests won't appear in the profiler.

http package (v1.2.2) does use `dart:io`'s `HttpClient` underneath, and the requests are still not shown. I even tried using cronet_http package, and its the same issue. It worked fine 2 or 3 months ago.

**Comment by chiphan-quickcare** [comment]
how to solve this?

**Comment by philitell** [comment]
Still facing the same issue with:
Flutter 3.24.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision dec2ee5c1f (vor 4 Wochen) • 2024-11-13 11:13:06 -0800
Engine • revision a18df97ca5
Tools • Dart 3.5.4 • DevTools 2.37.3

Requests are performed using Dio 5.2.0

**Comment by maxfrees** [comment]
This issue has been around for a long time, it seems to have been fixed with 3.24.0 before, and now it is happening again

**Comment by petersnoopy** [comment]
Still facing the same issue with:

Flutter 3.27.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 8495dee1fd (2 days ago) • 2024-12-10 14:23:39 -0800
Engine • revision 83bacfc525
Tools • Dart 3.6.0 • DevTools 2.40.2

**Comment by zigapovhe** [comment]
I can confirm after upgrading Flutter to 3.27.0, the HTTP requests are now visible in the DevTools network tab.
I've tried with http package, dio package and cronet_http package and all requests can be visible in the DevTools now.

    • Flutter version 3.27.0 on channel stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 8495dee1fd (2 days ago), 2024-12-10 14:23:39 -0800
    • Engine revision 83bacfc525
    • Dart version 3.6.0
    • DevTools version 2.40.2

To those still not seeing the requests after upgrading the Flutter, make sure DevTools version is 2.40.2 or higher.

**Comment by maxfrees** [comment]
> I can confirm after upgrading Flutter to 3.27.0, the HTTP requests are now visible in the DevTools network tab. I've tried with http package, dio package and cronet_http package and all requests can be visible in the DevTools now.
> 
> ```
> • Flutter version 3.27.0 on channel stable
> • Upstream repository https://github.com/flutter/flutter.git
> • Framework revision 8495dee1fd (2 days ago), 2024-12-10 14:23:39 -0800
> • Engine revision 83bacfc525
> • Dart version 3.6.0
> • DevTools version 2.40.2
> ```
> 
> To those still not seeing the requests after upgrading the Flutter, make sure DevTools version is 2.40.2 or higher.
Even if it's not 3.27.0, devtools will occasionally display http requests, and sometimes it won't display at all, showing only sockets

**Comment by Adrek** [comment]
> I can confirm after upgrading Flutter to 3.27.0, the HTTP requests are now visible in the DevTools network tab.
> I've tried with http package, dio package and cronet_http package and all requests can be visible in the DevTools now.

I'm using 3.27.0 and DevTools 2.40.2.
I tried with http and dio, but it doesn't work, it only shows SOCKET requests.

:(

**Comment by petersnoopy** [comment]
It seems to work for me now that I have deleted the cookies and website data.

**Comment by Adrek** [comment]
> It seems to work for me now that I have deleted the cookies and website data.

I tried the same thing as you, unfortunately it didn't work. I even tried it in a newly installed browser and it didn't work: Firefox.
I also tried it in the Network tab of Visual Code and the result was the same.

**Comment by mikegitdev** [comment]
Issue persists with Flutter 3.27.1, dart 3.6.0, dev tools 2.40.2.
Decided to switch back to 3.22.2 stable version. Let's wait for updates, otherwise SDK conflicts are real pain.

**Comment by Gastongouveia** [comment]
I realized what is causing the problem, in my case. Run a project that uses version 3.7.8 of flutter and then run one with version 3.27.0 on the same Android emulator. One solution is to cold boot the emulator when switching to the latest project. I'm using windows.

**Comment by kostas95** [comment]
Just like @Gastongouveia recommended, i cold booted my emulator on Android Studio too and the requrests showed up (using all latest versions)

**Comment by KazantsevLev** [comment]
> I realized what is causing the problem, in my case. Run a project that uses version 3.7.8 of flutter and then run one with version 3.27.0 on the same Android emulator. One solution is to cold boot the emulator when switching to the latest project. I'm using windows.

я твои губы целовал

**Comment by KarimMok** [comment]
Try to change emulator or build on real device i fixed the problème with this solutions.

**Comment by nishantpuri10** [comment]
https://github.com/flutter/devtools/issues/8798 , issue still persists even after using a new emulator and cold booting it everytime , actually it is an inconsistent bug , ie it sometimes work but most of the times shows only socket , also before doing flutter upgrade it use to work perfectly .

**Comment by srawlins** [comment]
Without the details requested [here](https://github.com/flutter/devtools/issues/8380#issuecomment-2392346295) and [here](https://github.com/flutter/devtools/issues/8380#issuecomment-2393929155), we cannot answer further. [Ben's answer](https://github.com/flutter/devtools/issues/8380#issuecomment-2393929155) is probably the answer.

**Comment by bkonyi** [comment]
I think I've found the cause of this issue and uploaded a fix in https://github.com/flutter/devtools/pull/8860.

TL;DR: it looks like there's a possible race condition that can cause HTTP requests to be missed by DevTools, particularly when the network channel between DevTools and the target application is slow. This is likely why some users seem to encounter the issue consistently where others can't reproduce it.

I hope to land the above PR and get it cherry picked into the next stable release.

**Comment by nishantpuri10** [comment]
Is there any method with which we can downgrade only the DevTools 2.40.2 , as it use to work perfectly in previous version ?

**Comment by bkonyi** [comment]
> Is there any method with which we can downgrade only the DevTools 2.40.2 , as it use to work perfectly in previous version ?

No, the only way to downgrade DevTools versions is to downgrade your Flutter/Dart SDKs. Hopefully we can get this fix cherry picked into the upcoming stable release so the next stable version of DevTools will include the fix.

**Comment by Mamasodikov** [comment]
Still the issue persists:

Flutter 3.27.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 17025dd882 (2 months ago) • 2024-12-17 03:23:09 +0900
Engine • revision cb4b5fff73
Tools • Dart 3.6.0 • DevTools 2.40.2

**Comment by hesptech** [comment]
one of the only ways to deal with this mess that Flutter mede in the release is to use Chrome as a DEVICE and in the browser DEVTOOL's debug ......
this problem comes a long way back and is not consistent...

**Comment by hesptech** [comment]
thanks!

Il giorno mar 4 mar 2025 alle ore 17:49 Kenzie Davisson <
***@***.***> ha scritto:

**Comment by kenzieschmoll** [comment]
Flutter 3.29.2 has been released to the Flutter stable channel. This Flutter version contains DevTools 2.42.3 with the fix for this issue. Please run `flutter upgrade` to get the latest Flutter and DevTools.

**Comment by alceil** [comment]
Im still getting issues

**Comment by BaroneX** [comment]
I have the same issue, I don't know how it became like this. It was fine in the morning, but suddenly stopped working in the afternoon.Don't just tell me to upgrade. I want to know the cause of the problem. It can't have suddenly stopped working for no reason.

Flutter (Channel stable, 3.24.5, on macOS 15.4.1 24E263 darwin-arm64, locale zh-Hans-CN)
    • Flutter version 3.24.5 on channel stable at /Users/blake/fvm/versions/3.24.5
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision dec2ee5c1f (9 months ago), 2024-11-13 11:13:06 -0800
    • Engine revision a18df97ca5
    • Dart version 3.5.4
    • DevTools version 2.37.3
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

**Comment by artisan-lai-07** [comment]
> I have the same issue, I don't know how it became like this. It was fine in the morning, but suddenly stopped working in the afternoon.Don't just tell me to upgrade. I want to know the cause of the problem. It can't have suddenly stopped working for no reason.
> 
> Flutter (Channel stable, 3.24.5, on macOS 15.4.1 24E263 darwin-arm64, locale zh-Hans-CN) • Flutter version 3.24.5 on channel stable at /Users/blake/fvm/versions/3.24.5 • Upstream repository https://github.com/flutter/flutter.git • Framework revision dec2ee5c1f (9 months ago), 2024-11-13 11:13:06 -0800 • Engine revision a18df97ca5 • Dart version 3.5.4 • DevTools version 2.37.3 • Pub download mirror https://pub.flutter-io.cn • Flutter download mirror https://storage.flutter-io.cn

I'm facing this issue as well in Flutter version `3.24.5`, any chance to get the DevTool update without upgrading the Flutter version?

**Comment by srawlins** [comment]
@alceil what version of Flutter are you running?

@artisan-lai-07 @BaroneX 

> Flutter 3.29.2 has been released to the Flutter stable channel. This Flutter version contains DevTools 2.42.3 with the fix for this issue. Please run flutter upgrade to get the latest Flutter and DevTools.
